From 377ad0a146899281b99641f5c61a4dbb978636c0 Mon Sep 17 00:00:00 2001
From: Christophe Parant <c.parant@phytec.fr>
Date: Mon, 16 Sep 2024 12:33:44 +0200
Subject: [PATCH 3/4] ARM v2021.10 stm32mp phy3 BOARD

---
 board/st/common/Kconfig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/board/st/common/Kconfig b/board/st/common/Kconfig
index 12abace091..f00168dd92 100644
--- a/board/st/common/Kconfig
+++ b/board/st/common/Kconfig
@@ -10,7 +10,7 @@ config MTDPARTS_NAND0_BOOT
 	string "mtd boot partitions for nand0"
 	default "2m(fsbl),2m(ssbl1),2m(ssbl2)" if STM32MP15x_STM32IMAGE || \
 						  !TFABOOT
-	default "512k(fsbl1),512k(fsbl2),512k(metadata1),512k(metadata2),4m(fip-a1),4m(fip-a2),4m(fip-b1),4m(fip-b2)"
+	default "2m(fsbl),4m(fip1),4m(fip2)"
 	depends on SYS_MTDPARTS_RUNTIME && ARCH_STM32MP
 	help
 	  This define the partitions of nand0 used to build mtparts dynamically
@@ -34,7 +34,7 @@ config MTDPARTS_NOR0_BOOT
 	string "mtd boot partitions for nor0"
 	default "256k(fsbl1),256k(fsbl2),2m(ssbl),512k(u-boot-env)" if STM32MP15x_STM32IMAGE || \
 								       !TFABOOT
-	default "256k(fsbl1),256k(fsbl2),256k(metadata1),256k(metadata2),4m(fip-a),4m(fip-b),512k(u-boot-env)"
+	default "256k(fsbl1),256k(fsbl2),4m(fip),512k(u-boot-env)"
 	depends on SYS_MTDPARTS_RUNTIME && ARCH_STM32MP
 	help
 	  This define the partitions of nand0 used to build mtparts dynamically
-- 
2.25.1

