Index: genparse-0.9.2/configure.ac
===================================================================
--- genparse-0.9.2.orig/configure.ac	2013-10-30 19:43:29.000000000 +0100
+++ genparse-0.9.2/configure.ac	2014-09-16 15:25:38.083688572 +0200
@@ -49,12 +49,7 @@
 AC_CHECK_FUNCS(mktime gethostname gettimeofday strdup strerror uname getdomainname getopt_long)
 
 dnl check for java Getopt class
-AC_CHECK_CLASS(gnu.getopt.Getopt)
-if test "x$ac_cv_class_gnu_getopt_Getopt" = xyes ; then
-        AC_SUBST(HAVE_GETOPT_JAVA, "yes")
-else
-	AC_MSG_WARN([Java class gnu.getopt.Getopt not found. You can build genparse anyway but not its test suite for the java output.])
-fi
+AC_MSG_WARN([Java class gnu.getopt.Getopt not found. You can build genparse anyway but not its test suite for the java output.])
 AM_CONDITIONAL(HAVE_GETOPT_JAVA, test "$HAVE_GETOPT_JAVA")
 
 dnl check for cunit
@@ -119,7 +114,5 @@
 
 gl_XSTRTOL
 
-AC_CONFIG_FILES([Makefile gnulib/Makefile gnulib/lib/Makefile src/Makefile doc/Makefile \
-	examples/Makefile tests/Makefile tests/ccheck/Makefile tests/cppcheck/Makefile
-	tests/javacheck/Makefile tests/misc/Makefile doc/Doxyfile])
+AC_CONFIG_FILES([Makefile gnulib/Makefile gnulib/lib/Makefile src/Makefile])
 AC_OUTPUT
Index: genparse-0.9.2/Makefile.am
===================================================================
--- genparse-0.9.2.orig/Makefile.am	2011-12-18 14:32:57.000000000 +0100
+++ genparse-0.9.2/Makefile.am	2014-09-16 15:26:37.335686702 +0200
@@ -1,5 +1,5 @@
 EXTRA_DIST = genparse.gp
 
-SUBDIRS = gnulib src doc examples tests
+SUBDIRS = gnulib src
 
 ACLOCAL_AMFLAGS = -I gnulib/m4
