From bf113361f763d787e91c3ee308f5658d37706051 Mon Sep 17 00:00:00 2001
From: Stefan Riedmueller <s.riedmueller@phytec.de>
Date: Thu, 17 Feb 2022 16:32:00 +0100
Subject: [PATCH] mediacontrol: V4L2File: Try both video devices

Signed-off-by: Stefan Riedmueller <s.riedmueller@phytec.de>
---
 mediacontrol/buffer/V4l2File.cpp | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/mediacontrol/buffer/V4l2File.cpp b/mediacontrol/buffer/V4l2File.cpp
index 563546be9a1b..977c0d7ff1ce 100644
--- a/mediacontrol/buffer/V4l2File.cpp
+++ b/mediacontrol/buffer/V4l2File.cpp
@@ -41,9 +41,16 @@ int V4l2File::open() {
         if (fd >= 0) {
             int streamid = -1;
             ioctl(fd, VIV_VIDIOC_S_STREAMID, &streamid);
-        } else
-            ALOGE("Failed to open video device(0)!\n");
+            return fd;
+        }
+        fd = openVideoDevice(1);
+        if (fd >= 0) {
+            int streamid = -1;
+            ioctl(fd, VIV_VIDIOC_S_STREAMID, &streamid);
+            return fd;
+        }
     }
+    ALOGE("Failed to open both video devices!\n");
     return fd;
 }
 
-- 
2.25.1

