From 93a2f68f791a0d3906d2a8068ce073168c01a109 Mon Sep 17 00:00:00 2001
From: Christophe Parant <c.parant@phytec.fr>
Date: Tue, 21 Feb 2023 09:47:26 +0100
Subject: [PATCH 1/2] v2.6 stm32mp phy1 MMC

---
 drivers/mmc/mmc.c | 18 ++++--------------
 1 file changed, 4 insertions(+), 14 deletions(-)

diff --git a/drivers/mmc/mmc.c b/drivers/mmc/mmc.c
index 2671bf4ee..736d67505 100644
--- a/drivers/mmc/mmc.c
+++ b/drivers/mmc/mmc.c
@@ -19,6 +19,7 @@
 
 #define MMC_DEFAULT_MAX_RETRIES		5
 #define SEND_OP_COND_MAX_RETRIES	100
+#define DEFAULT_CMD6_TIMEOUT_MS		500
 
 #define MULT_BY_512K_SHIFT		19
 
@@ -144,6 +145,7 @@ static int mmc_send_part_switch_cmd(unsigned int part_config)
 static int mmc_set_ext_csd(unsigned int ext_cmd, unsigned int value)
 {
 	int ret;
+	int timeout_ms = DEFAULT_CMD6_TIMEOUT_MS;
 
 	ret = mmc_send_cmd(MMC_CMD(6),
 			   EXTCSD_WRITE_BYTES | EXTCSD_CMD(ext_cmd) |
@@ -153,6 +155,8 @@ static int mmc_set_ext_csd(unsigned int ext_cmd, unsigned int value)
 		return ret;
 	}
 
+	mdelay(timeout_ms);
+
 	do {
 		ret = mmc_device_state();
 		if (ret < 0) {
@@ -284,13 +288,6 @@ static int mmc_fill_device_info(void)
 			return ret;
 		}
 
-		do {
-			ret = mmc_device_state();
-			if (ret < 0) {
-				return ret;
-			}
-		} while (ret != MMC_STATE_TRAN);
-
 		nb_blocks = (mmc_ext_csd[CMD_EXTCSD_SEC_CNT] << 0) |
 			    (mmc_ext_csd[CMD_EXTCSD_SEC_CNT + 1] << 8) |
 			    (mmc_ext_csd[CMD_EXTCSD_SEC_CNT + 2] << 16) |
@@ -545,13 +542,6 @@ static int mmc_enumerate(unsigned int clk, unsigned int bus_width)
 		return ret;
 	}
 
-	do {
-		ret = mmc_device_state();
-		if (ret < 0) {
-			return ret;
-		}
-	} while (ret != MMC_STATE_TRAN);
-
 	ret = mmc_set_ios(clk, bus_width);
 	if (ret != 0) {
 		return ret;
-- 
2.25.1

