From dc6bf3a1970bbf24f9112cd46847caa08e0f5e9f Mon Sep 17 00:00:00 2001
From: Christophe Parant <c.parant@phytec.fr>
Date: Tue, 31 May 2022 17:25:34 +0200
Subject: [PATCH] v2.4 stm32mp phy3 REGULATOR

---
 drivers/regulator/regulator_core.c | 21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/drivers/regulator/regulator_core.c b/drivers/regulator/regulator_core.c
index 67da7d292..b60e57fd9 100644
--- a/drivers/regulator/regulator_core.c
+++ b/drivers/regulator/regulator_core.c
@@ -321,6 +321,10 @@ int regulator_disable(struct rdev *rdev)
 
 	assert(rdev != NULL);
 
+	if (rdev->flags & REGUL_ALWAYS_ON) {
+		return 0;
+	}
+
 	ret = __regulator_set_state(rdev, STATE_DISABLE);
 
 	udelay(rdev->enable_ramp_delay);
@@ -711,6 +715,21 @@ static void parse_low_power_modes(const void *fdt, struct rdev *rdev, int node)
 		}
 	}
 }
+#else
+static int parse_properties(const void *fdt, struct rdev *rdev, int node)
+{
+	int ret;
+
+	if (fdt_getprop(fdt, node, "regulator-always-on", NULL) != NULL) {
+		VERBOSE("%s: set regulator-always-on\n", rdev->desc->node_name);
+		ret = regulator_set_flag(rdev, REGUL_ALWAYS_ON);
+		if (ret != 0) {
+			return ret;
+		}
+	}
+
+	return 0;
+}
 #endif
 
 /*
@@ -777,12 +796,12 @@ static int parse_dt(struct rdev *rdev, int node)
 		return ret;
 	}
 
-#if defined(IMAGE_BL32)
 	ret = parse_properties(fdt, rdev, node);
 	if (ret != 0) {
 		return ret;
 	}
 
+#if defined(IMAGE_BL32)
 	parse_supply(fdt, rdev, node);
 
 	parse_low_power_modes(fdt, rdev, node);
-- 
2.25.1

